### PR TITLE
[terra-content-container] Fixing the border appearing on click

### PR DESCRIPTION
**Pull request by AC086323**
### Summary

**What was changed:**
Added a condition for not setting the border when setFocusOnContainer is not there

https://github.com/cerner/terra-core/assets/126561169/fe4c0983-e3c5-406d-8b43-a047b6885e51

**Why it was changed:**
Because of the border appearing on click even though setFocusOnContainer is not there, this is breaking existing workflows and this fix needs to be addressed as soon as possible

### Testing

This change was tested using:

- [x] WDIO
- [ ] Jest
- [ ] Visual testing (please attach a screenshot or recording)
- [ ] Other (please describe below)
- [ ] No tests are needed

### Reviews

In addition to engineering reviews, this PR needs:

- [ ] UX review
- [ ] Accessibility review
- [ ] Functional review

### Additional Details

**This PR resolves:**

UXPLATFORM-XXXX 

---

Thank you for contributing to Terra.
@cerner/terra

### CODE REVIEW

File: packages/terra-content-container/src/ContentContainer.jsx [hunk 1]
```diff
@@ -68,7 +68,8 @@ const ContentContainer = ({
     customProps.className,
   ]);
 
-  const background = backgroundColor || 'light';
+  // border will only be visible when setFocusOnContainer is set for content-container
+  const background = setFocusOnContainer && (backgroundColor || 'light');
```
Comment (ShettyAkarsh) [inline packages/terra-content-container/src/ContentContainer.jsx:72]: what does this string 'light' signifies ?

Comment (AC086323) [reply packages/terra-content-container/src/ContentContainer.jsx:72]: It is used to default the background color to light if the backgroundColor prop is not set

File: packages/terra-content-container/CHANGELOG.md [hunk 2]
```diff
@@ -8,6 +8,9 @@
    * Added `setFocusOnContainer` prop to provide scroll capability for keyboard only users. (Requires Jest Snapshot updates)
    * Added `borderColor` prop for allowing consumers to change the border color to dark or white depending upon the color of the content container 
 
+* Changed
```
Comment (supreethmr) [inline packages/terra-content-container/CHANGELOG.md:11]: this change log should be moved to Unreleased section ( above the released version )

Comment (AC086323) [reply packages/terra-content-container/CHANGELOG.md]: Updated

File: packages/terra-content-container/src/ContentContainer.jsx (hunk 1 above)
Comment (ShettyAkarsh) [reply packages/terra-content-container/src/ContentContainer.jsx:72]: can you add to constant. can we keep the name as lightbackground ?

Comment (supreethmr) [reply packages/terra-content-container/src/ContentContainer.jsx]: We need to add a WDIO test here to ensure on performing click on a container focus is not added !!

**Comment by sdadn** [comment]
Is this a bug fix?

File: packages/terra-content-container/CHANGELOG.md [hunk 3]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Addressed the focus style on click issue when `setFocusOnContainer` is not set
```
Comment (sdadn) [inline packages/terra-content-container/CHANGELOG.md:6]:
```suggestion
  * Updated focus style when `setFocusOnContainer` is not set.
```

File: packages/terra-content-container/src/ContentContainer.jsx [hunk 4]
```diff
@@ -62,13 +62,15 @@ const ContentContainer = ({
   ...customProps
 }) => {
   const theme = React.useContext(ThemeContext);
+  const lightbackground = 'light';
```
Comment (sdadn) [inline packages/terra-content-container/src/ContentContainer.jsx:65]: nit: casing
```suggestion
  const lightBackground = 'light';
```

File: packages/terra-content-container/src/ContentContainer.jsx (hunk 1 above)
Comment (AC086323) [reply packages/terra-content-container/src/ContentContainer.jsx]: Updated

File: packages/terra-content-container/CHANGELOG.md [hunk 5]
```diff
@@ -2,6 +2,9 @@
 
 ## Unreleased
 
+* Changed
+  * Updated focus style when `setFocusOnContainer` is not set.
```
Comment (supreethmr) [inline packages/terra-content-container/CHANGELOG.md:6]:
```suggestion
* Fixed
  * Fixed issue of focus outline when `setFocusOnContainer` is not set.
```

**Review by supreethmr — APPROVED** [review]
+1 with change log update

**Pull request merged**